source/common/upstream/load_balancer_impl.h

```diff
@@ -454,9 +456,25 @@ class LeastRequestLoadBalancer : public EdfLoadBalancerBase {
     initialize();
   }
 
+  HostConstSharedPtr chooseHost(LoadBalancerContext* context) override {
+    alternativeWeights_ = Runtime::runtimeFeatureEnabled(
+        "envoy.reloadable_features.alternative_least_request_weights");
```

Comment:
For perf reasons I would not re-snap this on every choice. I would just do it in refresh below. I think this is fine.